internal/backend/discussion.go

```diff
@@ -475,6 +561,56 @@ func (d *delphisBackend) grantAccessAndCreateParticipants(ctx context.Context, d
 	return nil
 }
 
+func (d *delphisBackend) anonymizePostsForArchive(ctx context.Context, posts []*model.Post, shuffleCount int) ([]*model.ArchivedPost, error) {
+	archivedPosts := make([]*model.ArchivedPost, 0)
+
+	for _, post := range posts {
+		// Skip deleted posts for archive
+		if post.DeletedAt != nil {
+			continue
+		}
+
+		// Generate anonymous participant name
+		if post.DiscussionID == nil || post.ParticipantID == nil {
+			logrus.Error("discussion, participant should not be null")
+			continue
+		}
+
+		hashAsInt64 := util.GenerateParticipantSeed(*post.DiscussionID, *post.ParticipantID, shuffleCount)
+		fullDisplayName := util.GenerateFullDisplayName(hashAsInt64)
+
+		entities := make([]string, 0)
+		if post.PostContent.MentionedEntities != nil {
+			for _, entityID := range post.PostContent.MentionedEntities {
+				entity, err := util.ReturnParsedEntityID(entityID)
+				if err != nil {
+					logrus.WithError(err).Error("failed to parse entity")
+					return nil, err
+				}
+				if entity.Type == model.ParticipantPrefix {
+					hashAsInt64 := util.GenerateParticipantSeed(*post.DiscussionID, entity.ID, shuffleCount)
+					entities = append(entities, util.GenerateFullDisplayName(hashAsInt64))
+				} else if entity.Type == model.DiscussionPrefix {
+					entities = append(entities, "redacted_discussion")
```

Comment:
I think a mentioned discussion is actually okay here, no?

Reply:
what do we want the discussion to be? ID or title?

 I envisioned the "show convo on website" usecase and didn't know if the discussion made sense to link. happy to update it

Reply:
discussed offline and will keep it as `redacted`